microcosm_elasticsearch/tests/test_store.py

```diff
@@ -0,0 +1,201 @@
+"""
+Test Elasticsearch persistence.
+
+"""
+from hamcrest import (
+    all_of,
+    assert_that,
+    calling,
+    contains,
+    equal_to,
+    has_property,
+    is_,
+    none,
+    raises,
+)
+from microcosm.api import create_object_graph
+
+from microcosm_elasticsearch.assertions import assert_that_eventually, assert_that_not_eventually
+from microcosm_elasticsearch.errors import (
+    ElasticsearchConflictError,
+    ElasticsearchNotFoundError,
+)
+from microcosm_elasticsearch.indexing import createall
+from microcosm_elasticsearch.tests.fixtures import Person
+
+
+class TestStore(object):
+
+    def setup(self):
+        self.graph = create_object_graph("example", testing=True)
+        self.store = self.graph.person_store
+        createall(self.graph.example_index, force=True)
+
+        self.kevin = Person(
+            first="Kevin",
+            last="Durant",
+        )
+        self.steph = Person(
+            first="Steph",
+            last="Curry",
+        )
+
+    def test_retrieve_not_found(self):
+        assert_that(
+            calling(self.store.retrieve).with_args(self.store.new_object_id()),
+            raises(ElasticsearchNotFoundError),
+        )
+
+    def test_create(self):
+        self.store.create(self.kevin)
+        assert_that(
+            self.store.retrieve(self.kevin.id),
+            all_of(
+                has_property("id", self.kevin.id),
+                has_property("first", "Kevin"),
+                has_property("middle", none()),
+                has_property("last", "Durant"),
+            ),
+        )
+
+    def test_create_duplicate(self):
+        self.store.create(self.kevin)
+        assert_that(
+            calling(self.store.create).with_args(self.kevin),
+            raises(ElasticsearchConflictError),
+        )
+
+    def test_count(self):
+        self.store.create(self.kevin)
+        self.store.create(self.steph)
+        assert_that_eventually(
```

Comment:
`count` (and `search`) are queries and will not immediately read our writes.
`get` and `delete` do not have delays.